test/update_handler.c

```diff
@@ -261,9 +261,28 @@ static void test_update_handler(UpdateHandlerFixture *fixture,
 	if (!test_running_as_root())
 		return;
 
+	g_test_message("installing '%s' image to '%s' slot", test_pair->imagetype, test_pair->slottype);
+
 	/* prepare image and slot information */
 	imagename = g_strconcat("image.", test_pair->imagetype, NULL);
-	slotpath = g_build_filename(fixture->tmpdir, "rootfs-0", NULL);
+	if (g_strcmp0(test_pair->slottype, "ubivol") == 0||
+	    g_strcmp0(test_pair->slottype, "ubifs") == 0) {
+		slotpath = g_strdup(g_getenv("RAUC_TEST_MTD_UBIVOL"));
+		if (!slotpath) {
+			g_test_message("no UBI volume for testing found (define RAUC_TEST_MTD_UBIVOL)");
+			g_test_skip("RAUC_TEST_MTD_UBIVOL undefined");
+			return;
+		}
+	} else if (g_strcmp0(test_pair->slottype, "nand") == 0) {
+		slotpath = g_strdup(g_getenv("RAUC_TEST_MTD_NAND"));
+		if (!slotpath) {
+			g_test_message("no MTD NAND device for testing found (define RAUC_TEST_MTD_NAND)");
+			g_test_skip("RAUC_TEST_MTD_NAND undefined");
```

Comment:
`RAUC_TEST_MTD_NAND` is set unconditionally in `qemu-test-init`, this should get a simliar check as we have for `RAUC_TEST_MTD_UBI` already. Otherwise it fails if `flash_erase` (from mtd-utils) is not present.